pynxtools/dataconverter/readers/json_map/reader.py

```diff
@@ -152,6 +156,129 @@ def get_map_from_partials(partials, template, data):
     return mapping
 
 
+def parse_strings(mapping, data):
+    """
+    Parse strings, notably date and time, from custom format
+
+    The function can do the following operations, in the given order, on string data.
+    The result of each operation is passed on as input of the next one.
+
+    1. Extract element from array by index.
+    2. Match a regular expression.
+    3. Parse date and time using the datetime or dateutil parser.
+
+    The resulting string replaces the mapped value (dictionary) in the mapping dictionary.
+    If date parsing is enabled, the resulting string is ISO-formatted as required by the Nexus standard.
+    The operations are selected and tuned by the following dictionary items:
+
+    "parse_string": (required) Data path of the string (array) like for regular datasets.
+        If this item is missing, string parsing is skipped altogether.
+    "index": (optional) Element index to extract from string array.
+        The original data must be a string array.
+        If this option is not specified, the original data must be a singular string.
+    "regexp": (optional) Match regular expression, keeping only the matching part.
+        If the expression contains groups, the result will be a space-delimited concatenation of the matching groups.
+        If the expression does not contain explicit groups, the whole match is used.
+    "datetime": (optional) Format string for datetime.datetime.strptime function.
+        The "datetime" and "dateutil" options are mutually exclusive.
+    "dateutil": (optional) Date ordering for the dateutil.parser.parse function.
+        Possible values 'YMD', 'MDY', 'DMY' (or lower case).
+        The dateutil parsers recognizes many date and time formats, but may need the order of year, month and day.
+        The "datetime" and "dateutil" options are mutually exclusive.
+    "timestamp": (optional) Interpret the data item as POSIX timestamp.
+    "timezone": (optional) Specify the time zone if the date-time string does not include a UTC offset.
+        The time zone must be in a dateutil-supported format, e.g. "Europe/Berlin".
+        By default, the local time zone is used.
+    """
+
+    for key in mapping:
+        parse_opts = mapping[key]
+
+        try:
+            value = parse_opts["parse_string"]
+            if is_path(value):
+                value = get_val_nested_keystring_from_dict(value[1:], data)
+        except (KeyError, TypeError):
+            continue
+
+        if "index" in parse_opts:
+            value = value[int(parse_opts["index"])]
+
+        if "regexp" in parse_opts:
+            match = re.match(parse_opts["regexp"], value)
+            groups = match.groups('')
```

Comment:
Just for consistency you can do this:
```suggestion
            groups = match.groups("")
```